ebos/eclproblem.hh

```diff
@@ -2081,24 +2068,50 @@ public:
         serializer(tracerModel_);
         serializer(*materialLawManager_);
         serializer(*eclWriter_);
+    }    
+private:
+    Implementation& asImp_()
+    { return *static_cast<Implementation *>(this); }
+    
+    void updateExplicitQuantitites_()
+    {
+        OPM_TIMEBLOCK(updateExplicitQuantities);
+        const bool invalidateFromMaxWaterSat = updateMaxWaterSaturation_();
+        const bool invalidateFromMinPressure = updateMinPressure_();
+
+        // update hysteresis and max oil saturation used in vappars
+        const bool invalidateFromHyst = updateHysteresis_();
+        const bool invalidateFromMaxOilSat = updateMaxOilSaturation_();
+
+        // the derivatives may have change
+        bool invalidateIntensiveQuantities
+            = invalidateFromMaxWaterSat || invalidateFromMinPressure || invalidateFromHyst || invalidateFromMaxOilSat;
+        if (invalidateIntensiveQuantities) {
+            OPM_TIMEBLOCK(beginTimeStepInvalidateIntensiveQuantities);
+            this->model().invalidateAndUpdateIntensiveQuantities(/*timeIdx=*/0);
+        }
+
+        if constexpr (getPropValue<TypeTag, Properties::EnablePolymer>())
+            updateMaxPolymerAdsorption_();
     }
 
-private:
     template<class UpdateFunc>
     void updateProperty_(const std::string& failureMsg,
                          UpdateFunc func)
     {
         OPM_TIMEBLOCK(updateProperty);
-        ElementContext elemCtx(this->simulator());
+        const auto& problem = this->simulator().problem();
```

Comment:
Tiny nit: This `problem` object is not used anywhere in the `updateProperty_()` function.  Is that intentional?  If so, I suggest we just remove it.

Reply:
> fixed

Excellent.  I'll just run final build check and then merge when the check is green.